biogrid/libs/biogrid-simulator/src/lib/biobattery/biobattery-class.ts

```diff
@@ -0,0 +1,59 @@
+import { Battery, Energy, Validatable, validate } from "@biogrid/grid-simulator";
+
+export class BioBattery implements Battery {
+  private currentBatteryPower: Energy;
+  private readonly maxCapacity: Energy;
+
+  constructor(currentBatteryPower: Energy = 0, maxCapacity: Energy = 10) {
+    if (!this.validateInputs(currentBatteryPower, maxCapacity)) {
+      throw new Error('Cannot create a battery with these values');
+    }
+    this.currentBatteryPower = currentBatteryPower;
+    this.maxCapacity = maxCapacity;
+  }
+
+  startCharging(inputPower: Energy): void {
+    if (this.currentBatteryPower + inputPower > this.maxCapacity) {
+      this.currentBatteryPower = this.maxCapacity;
+    }
+    this.currentBatteryPower += inputPower;
+  }
+
+  // TODO implement when you use a formula for charging a battery
+  stopCharging(): void { }
+
+  supplyPower(outputenergy: Energy): Energy {
+    if (this.currentBatteryPower - outputenergy < 0) {
+      throw new Error(`Battery has less than ${outputenergy} units`);
```

Comment:
Should the battery still provide as much energy as possible? 
Maybe a way of implementing this could be to have supplyPower return how much Energy it successfully supplied. So it might supply the full required energy, or it might supply a portion of it, in which case the caller of supplyPower would know to request more power from another energy source